packages/x-data-grid-pro/src/hooks/features/serverSideLazyLoader/useGridDataSourceLazyLoader.ts

```
@@ -195,11 +230,12 @@ export const useGridDataSourceLazyLoader = (
         rows: {
           ...state.rows,
           tree,
+          dataRowIdToModelLookup,
```

should we let the grid rebuild this?
or it degrades the performance too much?

---

> should we let the grid rebuild this?

The idea is to get rid of the removed nodes ASAP, this seems to be the best place since we are already looping over these nodes. Haven't measured the performance after off-loading this to the Grid but I'm afraid there will be the unattached nodes in `dataRowIdToModelLookup` until next full regeneration of the tree if we don't do it here.

---

> we want to access the data that is not there

If we are trying to access some data that belongs to the outdated rows, I think it'd be a bug and should be avoided, keeping these rows in state would silently suppress these errors which might end up in worse issues. Agree on leaving it as is.